File: app/src/main/java/home/oleg/popularmovies/presentation/list/ListView.java
package home.oleg.popularmovies.presentation.list;

import java.util.List;

import home.oleg.popularmovies.presentation.model.MovieViewModel;

/**
 * Created by Oleg on 15.04.2017.
 */

public interface ListView {
    void showLoading();
    void hideLoading();
    void showError();
    void fillList(List<MovieViewModel> models);
}

File: app/src/main/java/home/oleg/popularmovies/presentation/list/ListView.kt
package home.oleg.popularmovies.presentation.list

import home.oleg.popularmovies.presentation.base.WithError
import home.oleg.popularmovies.presentation.base.WithLoading
import home.oleg.popularmovies.presentation.model.MovieViewModel

/**
 * Created by Oleg on 15.04.2017.
 */

interface ListView : WithError, WithLoading {
    fun fillList(models: List<MovieViewModel>)
}
